shared_model/builders/protobuf/transaction.hpp

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_PROTO_TRANSACTION_BUILDER_HPP
+#define IROHA_PROTO_TRANSACTION_BUILDER_HPP
+
+#include "backend/protobuf/transaction.hpp"
+
+#include "block.pb.h"
+#include "commands.pb.h"
+
+#include "amount/amount.hpp"
+#include "builders/protobuf/helpers.hpp"
+#include "interfaces/common_objects/types.hpp"
+
+namespace shared_model {
+  namespace proto {
+    template <int S = 0>
+    class TemplateTransactionBuilder {
+     private:
+      template <int>
+      friend class TemplateTransactionBuilder;
+
+      enum RequiredFields { Command, CreatorAccountId, TxCounter, TOTAL };
+
+      template <int s>
+      using NextBuilder = TemplateTransactionBuilder<S | (1 << s)>;
+
+      iroha::protocol::Transaction transaction_;
+
+      template <int Sp>
+      TemplateTransactionBuilder(const TemplateTransactionBuilder<Sp> &o)
```

Comment:
Why is copy ctor private?

Reply:
Seems like merge problem, thanks for noticing